packages/nimble-components/src/number-field/index.ts

```diff
@@ -40,11 +42,67 @@ export class NumberField extends mixinErrorPattern(
     @attr({ attribute: 'appearance-readonly', mode: 'boolean' })
     public appearanceReadOnly = false;
 
+    private decimalSeparator = '.';
+    private inputFilterRegExp: RegExp | null = null;
+
+    private readonly langSubscriber: DesignTokenSubscriber<typeof lang> = {
+        handleChange: () => this.updateDecimalSeparatorAndInputFilter()
+    };
+
     public override connectedCallback(): void {
         super.connectedCallback();
 
         // This is a workaround for FAST issue: https://github.com/microsoft/fast/issues/6148
         this.control.setAttribute('role', 'spinbutton');
+        this.updateDecimalSeparatorAndInputFilter();
+        lang.subscribe(this.langSubscriber, this);
+    }
+
+    public override disconnectedCallback(): void {
+        super.disconnectedCallback();
+
+        lang.unsubscribe(this.langSubscriber, this);
+    }
+
+    protected override sanitizeInput(inputText: string): string {
+        return inputText.replace(this.inputFilterRegExp!, '');
```

Comment:
I think we should handle the case where `sanitizeInput` is called before `connectedCallback` rather than assuming `inputFilterRegExp` is non-null. It may be that can't happen right now given the FAST implementation (I didn't check) but that seems like an easy thing to forget to re-check if we ever modify the base class again.

One option would be to call `updateDecimalSeparatorAndInputFilter()` from a constructor. Or just set `inputFilterRegExp` to an initial value that assumes the `.` separator.

Reply:
Added private `buildFilterRegExp()` function to single-source the regex expression.